library/core/src/slice/mod.rs

```diff
@@ -1099,10 +1099,15 @@ impl<T> [T] {
     /// assert!(iter.next().is_none());
     /// ```
     ///
-    /// There's no `windows_mut`, as that existing would let safe code violate the
-    /// "only one `&mut` at a time to the same thing" rule.  However, you can sometimes
-    /// use [`Cell::as_slice_of_cells`](crate::cell::Cell::as_slice_of_cells) in
+    /// Because the [Iterator] trait cannot represent the required lifetimes, there is no `windows_mut` analog to `windows`.
+    /// (If there was, then `[0,1,2].windows_mut(2).collect()` would create 2 mutable references
+    /// to the middle element of that array, which would violate [the rules of references],
+    /// though a [LendingIterator] analog is possible.)
+    /// However, you can sometimes use [`Cell::as_slice_of_cells`](crate::cell::Cell::as_slice_of_cells) in
```

Comment:
I think things can be simplified a bit: complete thoughts shouldn't be in `( ... )`, and I don't think we need to point link to the reference docs (we can probably assume users know that two mutable references to the same thing is bad). So suggested wording, also dropping the `LendingIterator` mention:

```suggestion
    /// There is no `windows_mut` analog to `windows` because [`Iterator`] does not represent the
    /// required lifetimes, e.g. `[0, 1, 2].windows_mut(2).collect()` would allow creating two mutable
    /// references to the middle element of the array. However, you can sometimes use
    /// [`Cell::as_slice_of_cells`] (crate::cell::Cell::as_slice_of_cells) in
```

Then add something like "This pattern can also be achieved with a [`LendingIterator`]." after the example (to keep the paragraph introducing the example only about `std`).

Reply:
Regarding complete thought in parens: The second site I looked up gave this example: [Tonight's gala is a huge occasion. (Everyone who's somebody will be there.)](https://writingtips.org/parentheses/#give_additional_information), so I don't quite see what is wrong with my use of parens here. I've used the parens in the sense that you can get more information by reading what's inside them, but you can also safely skip over them, and this way the reference outside of std is not in the unparenthesized text. Upon further reflection I suppose I could move the parens to only go around "though a [LendingIterator] analog is possible".

Reply:
I've made the parenthetical more terse and adjusted punctuation. Let me know if you think it helps.

Reply:
It is grammatically correct, but readability tends to be better if parenthesized data is kept brief. If there are complete asides that need to be expressed, it is preferable to put it in a footnote or separate paragraph, or just fold it into the text somehow.

Reword looks fine though 👍

Reply:
Great!